src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala

```diff
@@ -0,0 +1,50 @@
+package beam.sim
+
+import beam.sim.config.BeamExecutionConfig
+import beam.utils.TestConfigUtils.testConfig
+import com.typesafe.config.ConfigFactory
+import org.matsim.core.scenario.MutableScenario
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.nio.file.Paths
+
+class WritePlansAndStopSimulationTest extends AnyFlatSpec with Matchers with BeamHelper {
+  it should "Stop simulation, trow runtime exception and write plans to the output folder." in {
+    val config = ConfigFactory
+      .parseString(s"""
+           |beam.agentsim.lastIteration = 3
+           |beam.outputs.events.fileOutputFormats = "xml,csv"
+           |beam.outputs.writePlansInterval = 3
+           |beam.physsim.writePlansInterval = 0
+           |beam.output.writePlansAndStopSimulation = true
+         """.stripMargin)
+      .withFallback(testConfig("test/input/beamville/beam.conf"))
+      .resolve()
+
+    val (
+      beamExecutionConfig: BeamExecutionConfig,
+      scenario: MutableScenario,
+      beamScenario: BeamScenario,
+      services: BeamServices,
+      plansMerged: Boolean
+    ) = prepareBeamService(config, None)
+
+    val exception: RuntimeException = intercept[RuntimeException] {
+      runBeam(
+        services,
+        scenario,
+        beamScenario,
+        beamExecutionConfig.outputDirectory,
+        plansMerged
+      )
+    }
+
+    exception.getMessage shouldBe "The simulation was stopped because beam.output.writePlansAndStopSimulation set to true."
```

Comment:
nit = I'm never a fan of a string comparison like this (brittle).

Reply:
Yes, generally I agree, but for this case we need to either check the error message or to throw a special type of error just for this case, which, in my opinion, is overkill.